src/services/createParam.ts

```diff
@@ -0,0 +1,18 @@
+import { getParamValue, setParamValue } from '@/services/params'
+import { ParamWithDefault } from '@/services/withDefault'
+import { ExtractParamType, isParamGetSet } from '@/types/params'
+import { Param, ParamGetSet } from '@/types/paramTypes'
+
+export function createParam<TParam extends Param>(param: TParam): ParamGetSet<ExtractParamType<TParam>>
+export function createParam<TParam extends Param>(param: TParam, defaultValue: ExtractParamType<TParam>): ParamWithDefault<TParam>
```

Comment:
Could we add another overload where TParam extends ParamWithDefault? So it could be defined as a single object.